File: Python/TextbookTrack/ba1g.py
# Compute the Hamming Distance Between Two Strings
import sys


def calc_hamming_distance(s1, s2):
    distance = 0
    for c1, c2 in zip(s1, s2):
        if c1 != c2:
            distance += 1
    return distance


def main():
    data = sys.stdin.read().splitlines()
    dna1 = data[0]
    dna2 = data[1]
    result = calc_hamming_distance(dna1, dna2)
    print(result)


if __name__ == '__main__':
    main()

File: Julia/textbook_track/chapter01/ba1g.jl
# Compute the Hamming Distance Between Two Strings

calc_hamming_distance(s1, s2) = count(((c1, c2),) -> c1 != c2, zip(s1, s2))


function main()
    dna1 = readline()
    dna2 = readline()
    result = calc_hamming_distance(dna1, dna2)
    println(result)
end


main()
